controllers/provisioning_controller.go

```diff
@@ -299,16 +297,47 @@ func (r *ProvisioningReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return result, nil
 }
 
-func (r *ProvisioningReconciler) provisioningInfo(provConfig *metal3iov1alpha1.Provisioning, images *provisioning.Images, proxy *osconfigv1.Proxy) *provisioning.ProvisioningInfo {
+func (r *ProvisioningReconciler) provisioningInfo(provConfig *metal3iov1alpha1.Provisioning, images *provisioning.Images) (*provisioning.ProvisioningInfo, error) {
+	clusterWideProxy, err := r.OSClient.ConfigV1().Proxies().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	ips, err := net.LookupIP(config.Host)
+	if err != nil {
+		return nil, err
+	}
+	haveV4 := false
+	haveV6 := false
+	for _, ip := range ips {
+		if len(ip) == net.IPv4len {
+			haveV4 = true
+		} else if len(ip) == net.IPv6len {
+			haveV6 = true
+		}
+	}
+	ns := provisioning.NetworkStackV4
+	if haveV4 && haveV6 {
+		ns = provisioning.NetworkStackDual
+	} else if haveV6 {
+		ns = provisioning.NetworkStackV6
+	}
```

Comment:
this seems really clumsy (and not tested), if anyone can think of a better way, please let me know...